erigon-lib/kv/kvcache/cache.go

```diff
@@ -334,40 +334,37 @@ func (c *Coherent) OnNewBlock(stateChanges *remote.StateChangeBatch) {
 }
 
 func (c *Coherent) View(ctx context.Context, tx kv.Tx) (CacheView, error) {
-	idBytes, err := tx.GetOne(kv.Sequence, kv.PlainStateVersion)
+	id, err := tx.ReadSequence(string(kv.PlainStateVersion))
 	if err != nil {
 		return nil, err
 	}
-	var id uint64
-	if len(idBytes) == 0 {
-		id = 0
-	} else {
-		id = binary.BigEndian.Uint64(idBytes)
-	}
+
+	c.lock.Lock() 
 	r := c.selectOrCreateRoot(id)
+	isReady := r.readyChanClosed.Load()
```

Comment:
1. `selectOrCreateRoot` is thread-safe
2. `readyChanClosed` i don't understand why need check it. also i don't understand why we need this atomic. seems better replace by `sync.Once`